Gifski/MainWindowController.swift

```diff
@@ -138,23 +173,39 @@ final class MainWindowController: NSWindowController {
 			)
 
 			Gifski.run(conversion) { error in
-				if let error = error {
-					fatalError(error.localizedDescription)
+				DispatchQueue.main.async {
+					self.isRunning = false
 				}
 
-				// Workaround for https://github.com/sindresorhus/gifski-app/issues/46
-				progress.completedUnitCount = 0
+				guard let error = error else {
+					return
+				}
 
-				DispatchQueue.main.async {
-					self.circularProgress.fadeOut(delay: 1) {
-						self.circularProgress.resetProgress()
-						self.isRunning = false
-					}
+				switch error {
+				case .cancelled:
+					break
+				default:
+					fatalError(error.localizedDescription)
 				}
 			}
 		}
 	}
 
+	private func onHover(_ event: HoverView.Event) {
+		switch event {
+		case .entered:
+			circularProgress.isProgressLabelHidden = true
+			cancelButton.fadeIn()
+		case .exited:
+			circularProgress.isProgressLabelHidden = false
+			cancelButton.isHidden = true
+		}
+	}
+
+	private func cancelConversion(_: NSControl?) {
```

Comment:
I don't think this should have an argument as it's used in multiple places:

```suggestion
	private func cancelConversion() {
```

Instead, on line 158, use:

```swift
cancelButton.onAction = { _ in
	self.cancelConversion()
}
```

Reply:
I can appreciate the brevity of simply assigning the method to the `onAction` property, but I've committed the change.

Reply:
It's actually less characters this way. Really depends on how you measure brevity.